### PR TITLE
chore: cascader updates

### PR DESCRIPTION
**Pull request by jared-dickman**
## Summary
- Add `placement` prop to QueryItem cascader
- change Label from a string to a react component, allowing for complex cascader items
- Add a searchLabel prop to incorporate searching on a complex label
- turn on autofocus when the cascader opens

### CODE REVIEW

**Comment by ilyusha** [comment]
### Review Forge
[Click Here for AI Insights](https://supreme-adventure-m612l2p.pages.github.io?prUrl=https://github.com/mParticle/aquarium/pull/378)

**Comment by nrmerlis** [comment]
"This branch has conflicts that must be resolved" @jared-dickman

File: src/components/data-entry/QueryItem/Cascader.tsx [hunk 1]
```diff
@@ -15,9 +16,10 @@ import { type Icons } from 'src/constants/Icons'
 import { useMount } from 'src/hooks/useMount'
 import { debounce } from 'src/utils/utils'
 
-export interface ICascaderOption {
+export interface ICascaderOption extends DefaultOptionType {
   value: string
-  label: string
+  label: ReactNode
+  searchLabel?: string // useful when label is a reactNode and not a string
```
Comment (danielsiemens) [inline src/components/data-entry/QueryItem/Cascader.tsx:22]: does that mean `label` should be of type `ReactNode | string`?

Comment (danielsiemens) [reply src/components/data-entry/QueryItem/Cascader.tsx:22]: Ahh ignore me. Forgot that `string` is a valid ReactNode

**Review by danielsiemens — APPROVED** [review]

File: src/components/data-entry/QueryItem/Cascader.tsx (hunk 1 above)
Comment (jared-dickman) [reply src/components/data-entry/QueryItem/Cascader.tsx:22]: actually strings are valid ReactNodes so its redundant, though more explicit 🤔

**Pull request merged**